### PR TITLE
Layout transform control

### PR DESCRIPTION
**Pull request by donandren**
Layout Transform control to enable Layout transformations of child control via Matrix transforms etc.

### CODE REVIEW

**Comment by grokys** [comment]
Thanks for this, I need to take a good look at it, but looking good so far!

File: src/Perspex.Controls/LayoutTransformControl.cs [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+            ApplyLayoutTransform();
+        }
+
+        /// <summary>
+        /// Applies the layout transform on the LayoutTransformerControl content.
+        /// </summary>
+        /// <remarks>
+        /// Only used in advanced scenarios (like animating the LayoutTransform).
+        /// Should be used to notify the LayoutTransformer control that some aspect
+        /// of its Transform property has changed.
```
Comment (grokys) [inline src/Perspex.Controls/LayoutTransformControl.cs]: When animating the layout transform, `OnLayoutTransformChanged` will be called, which will call this method. Does it really need to be public?

File: src/Perspex.Controls/LayoutTransformControl.cs [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+        private Size _childActualSize = Size.Empty;
+
+        private Control _transformRoot;
+
+        public Control TransformRoot => _transformRoot ??
+                    (_transformRoot = this.GetVisualChildren().OfType<Control>().FirstOrDefault());
+
+        private IDisposable _transformChangedEvent = null;
+
+        private void OnLayoutTransformChanged(PerspexPropertyChangedEventArgs e)
```
Comment (grokys) [inline src/Perspex.Controls/LayoutTransformControl.cs]: Nit: could you order methods public, protected, private?

File: src/Perspex.Controls/LayoutTransformControl.cs [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+        /// Acceptable difference between two doubles.
+        /// </summary>
+        private const double AcceptableDelta = 0.0001;
+
+        /// <summary>
+        /// Number of decimals to round the Matrix to.
+        /// </summary>
+        private const int DecimalsAfterRound = 4;
+
+        private static readonly FuncControlTemplate<LayoutTransformControl> _defaultTemplate =
```
Comment (grokys) [inline src/Perspex.Controls/LayoutTransformControl.cs]: No other controls have a default template expressed in code, probably best to just add a XAML template to Perspex.Styles.Default as with other controls.

File: src/Perspex.Controls/LayoutTransformControl.cs (hunk 1 above)
Comment (donandren) [inline src/Perspex.Controls/LayoutTransformControl.cs]: most probably not, so I'm making it private

File: src/Perspex.Controls/LayoutTransformControl.cs (hunk 2 above)
Comment (donandren) [inline src/Perspex.Controls/LayoutTransformControl.cs]: ready

File: src/Perspex.Controls/LayoutTransformControl.cs (hunk 3 above)
Comment (donandren) [inline src/Perspex.Controls/LayoutTransformControl.cs]: ready

**Comment by donandren** [comment]
@grokys  all recomendationa are implemented

**Comment by grokys** [comment]
Thanks - I'm about to go away for a few days, I will take a good look early next week.

**Comment by grokys** [comment]
Merged - apologies for the time taken on this, it was an excellent PR! I also added it to the ControlCatalog.

**Pull request merged**